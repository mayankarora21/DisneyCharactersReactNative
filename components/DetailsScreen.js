import { View, Image, ScrollView, StyleSheet } from "react-native";
import { useEffect } from "react";
import DetailItem from "./DetailItem";

const DetailsScreen = ({route, navigation}) => {
  const character = route.params.character;

  useEffect(() => {
    navigation.setOptions({ title: character.name });
  }, [navigation, character]);

  const keys = [
    { visibleKey: "Films", actualKey: "films" },
    { visibleKey: "Short Films", actualKey: "shortFilms" },
    { visibleKey: "TV Shows", actualKey: "tvShows" },
    { visibleKey: "Video Games", actualKey: "videoGames" },
    { visibleKey: "Park Attractions", actualKey: "parkAttractions" },
    { visibleKey: "Allies", actualKey: "allies" },
    { visibleKey: "Enemies", actualKey: "enemies" },
  ];

  return (
    <View style={styles.rootContainer}>
      <Image source={{ uri: character.imageUrl }} style={styles.image}></Image>
      <View style = {styles.detailsContainer}>
        <ScrollView>
          {
              keys.map((key, index) => {
                  return (
                      <DetailItem character={character} key = {index} keyObject = {key}></DetailItem>
                  );
              })
          }
        </ScrollView>
      </View>
    </View>
  );
};

export default DetailsScreen;

const styles = StyleSheet.create({
  rootContainer: {
    alignItems: "center",
    flex: 1
  },
  image: {
    height: 200,
    width: 200,
    resizeMode: "stretch",
    borderRadius: 100,
    marginVertical: 15
  },
  detailsContainer: {
    width: '90%',
    flex: 1,
    marginBottom: 15
  }
});