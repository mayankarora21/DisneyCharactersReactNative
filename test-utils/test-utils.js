import React from 'react'
import {render} from '@testing-library/react-native';
import store from '../store/store';
import { Provider } from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';

const AllTheProviders = ({children}) => {
  return (
    <Provider store = {store}>
      <NavigationContainer>
            {children}
      </NavigationContainer>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}