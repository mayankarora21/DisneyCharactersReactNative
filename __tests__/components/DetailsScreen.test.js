import {render, screen} from '../../test-utils/test-utils';
import DetailsScreen from '../../components/DetailsScreen';

test('DetailsScreen should render correctly', () => {
    render(<DetailsScreen route = {{
        params: {
            character: {
                name: 'Character Name',
                imageUrl: "https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png",
                films: ["Film1", "Film2"],
                shortFilms: [],
                tvShows: [],
                videoGames: [],
                parkAttractions: [],
                allies: [],
                enemies: [],
            }
    }}} navigation = {{setOptions: jest.fn()}}/>);
    expect(screen.toJSON()).toMatchSnapshot();
})