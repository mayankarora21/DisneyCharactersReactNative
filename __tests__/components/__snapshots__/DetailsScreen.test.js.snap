// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png",
      }
    }
    style={
      {
        "borderRadius": 100,
        "height": 200,
        "marginVertical": 15,
        "resizeMode": "stretch",
        "width": 200,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "marginBottom": 15,
        "width": "90%",
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          Films
          : 
          Film1, Film2
        </Text>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          Short Films
          : 
          No Short Films
        </Text>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          TV Shows
          : 
          No TV Shows
        </Text>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          Video Games
          : 
          No Video Games
        </Text>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          Park Attractions
          : 
          No Park Attractions
        </Text>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          Allies
          : 
          No Allies
        </Text>
        <Text
          style={
            {
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          Enemies
          : 
          No Enemies
        </Text>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
